config/mbed/chip-gn/mbedtls/BUILD.gn

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Options from standalone-chip.mk that differ from configure defaults. These
+# options are used from examples/.
+
+import("//build_overrides/chip.gni")
+
+declare_args() {
+  chip_with_platform_mbedtls = false
+}
+
+if (!chip_with_platform_mbedtls) {
+    import("//build_overrides/mbedtls.gni")
+    import("${mbedtls_root}/mbedtls.gni")
+
+    config("mbed_mbedtls_config") {
+        include_dirs = [
+            "."
+        ]
+        defines = [ 
+            "MBEDTLS_CONFIG_FILE=<mbed_mbedtls_config.h>"
```

Comment:
But it is a parameter used by internal CHIP mbedtls implementation. If we set it to mbed_app.json file it will be common to both options (CHIP mbedtls and mbed-os mbedtls). For now, we use the default config for mbedtls from mbed-os. Is it the goal?

Reply:
I think we can consider the other way: pointing here to the configuration file from mbed-os source. But it should be a directly path such as /opt/mbed-os/connectivity/mbedtls/include/config.h

Reply:
I have a better understanding of this now: 

- `MBEDTLS_CONFIG_FILE` should be pointing to the base configuration for CHIP it can enable the define we need like: `MBEDTLS_X509_CSR_WRITE_C`. The define should be available for the **builder** (mbed) and **consumer** (chip). 
- `MBEDTLS_USER_CONFIG_FILE` is defined by the application if it needs to override the base CHIP config. 

I'm not sure of what is the right approach today. I'm okay with this being defined here and in every `mbed_app.json` as a short term solution (we just have 1 today) and revisit that item later in the project. If we use a short term solution, we need a ticket for it so we can revisit in the future.